org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -2645,6 +2645,23 @@ public final class JavaCore extends Plugin {
 	 * @since 3.18
 	 */
 	public static final String COMPILER_PB_REPORT_PREVIEW_FEATURES = PLUGIN_ID + ".compiler.problem.reportPreviewFeatures"; //$NON-NLS-1$
+
+	/**
+	 * Compiler option ID: Ignore unnamed module for split package.
+	 * <p>
+	 * When enabled, the compiler will ignore the unnamed module when a package is defined in both a named module and the unnamed module.
+	 * This preference is introduced due to the explanation described in https://openjdk.org/projects/jigsaw/spec/sotms/#the-unnamed-module.
```

Comment:
I think without further context, this will leave readers confused. I'm afraid we'll have to explain a bit more, like:
```
	 * <p>
	 * With this option the compiler will deliberately accept programs violating JLS in a specific way.
	 * Instead the compiler will behave in accordance to the original, but unmaintained document
	 *  <a href="https://openjdk.org/projects/jigsaw/spec/sotms/#the-unnamed-module">"The State of the Module System"</a>,
	 * which indicates that different semantics had been intended.
	 * </p>
```